.github/workflows/main.yml

```diff
@@ -556,8 +556,9 @@ jobs:
   #         event-type: python-hil-event
   #         client-payload: '{"ref": "${{ github.ref }}", "sha": "${{ github.sha }}"}'
 
-  notify_hil_workflow_linux_x86_64:
-    needs: [build-linux-x86_64]
+  notify_hil_workflow:
+    if: '!cancelled() && success()' # run when jobs are skipped, but don't run when one fails
+    needs: [pytest, build-linux-armhf, build-windows-x86_64, build-macos-x86_64, build-macos-arm64, build-linux-x86_64, build-linux-arm64, deploy]
```

Comment:
Let's remove `deploy` here, since that is only executed when we do a release and we want to run tests on every PR & commit to develop.

Reply:
It will run on every commit as it is. The !cancelled() flag is similar to always() flag, it causes the job to run always even when its "needs" aren't met and previous jobs were skipped or failed (that is why there is the success() flag so we don't run hil workflow when some builds fail). We need to run it after deploy, because deploy uploads the wheel for tagged releases as tagged releases are not uploaded before in the pipeline (there is an if statement that disables the upload for tagged releases). You can see a test case here https://github.com/jakgra/jakgra-test-github-actions/actions/runs/7475941665/workflow

Reply:
The notify hil job now depends only on the builds it needs and not on deploy anymore.